neo/Network/P2P/TaskManager.cs

```diff
@@ -202,13 +202,13 @@ private void RequestTasks(TaskSession session)
                 session.AvailableTasks.ExceptWith(knownHashes);
                 session.AvailableTasks.RemoveWhere(p => Blockchain.Singleton.ContainsBlock(p));
                 HashSet<UInt256> hashes = new HashSet<UInt256>(session.AvailableTasks);
-                foreach (UInt256 hash in hashes.ToArray())
-                {
-                    if (!IncrementGlobalTask(hash))
-                        hashes.Remove(hash);
-                }
                 if (hashes.Count > 0)
                 {
+                    foreach (UInt256 hash in hashes.ToArray())
```

Comment:
In the minor change you made moving this in here after this loop hashes could be 0. It looks like nothing bad happens, but seem slightly more optimal leaving the loop outside the if. Why did you decide to move it?

Reply:
This may avoid `hashes.ToArray()`.

Reply:
I see, thanks for the explanation.